ls1intum/Artemis
# `Development`: Replace learning goal with competency in client file and class names

### Checklist
#### General



#### Client

### Motivation and Context


This is a part of follow-up PRs on https://github.com/ls1intum/Artemis/pull/6388 There's currently an inconsistency between the usage of 'learning goal' and 'competency' throughout the system. This PR **only addresses the client file and class names**. 

### Description

Replaced all occurrences of the term 'learning goal' with 'competency' on client file and class names. Also, some fields and functions are renamed to match the new class names. There's only one minor change to a server file that affects the label generation for alerts.

###### NOTE: Documentation files are not covered by this PR.

### Steps for Testing

Prerequisites:
- 1 Course with competencies
- 1 Admin

1. Log in to Artemis as Admin
2. Navigate to the Competency Management page of the course
3. Test CRUD operations (+import) for competencies
4. Link this competency to a Lecture Unity or Exercise and see if completing it updates the competency progress for the user

### Review Progress
#### Code Review
#### Manual Tests
